backend/domain/domainmodels.js

```diff
@@ -0,0 +1,119 @@
+class Room {
+    constructor() {
```

Comment:
I think the methods should be outside the constructor?

Reply:
changed along with changing all functions to format: functionname(){  }